src/routes/api/ollama.ts

```diff
@@ -0,0 +1,73 @@
+import axios from 'axios';
+import options from "../../services/options.js";
+import log from "../../services/log.js";
+import type { Request, Response } from "express";
+
+/**
+ * @swagger
+ * /api/ollama/models:
+ *   post:
+ *     summary: List available models from Ollama
+ *     operationId: ollama-list-models
+ *     requestBody:
+ *       required: false
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             properties:
+ *               baseUrl:
+ *                 type: string
+ *                 description: Optional custom Ollama API base URL
+ *     responses:
+ *       '200':
+ *         description: List of available Ollama models
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 success:
+ *                   type: boolean
+ *                 models:
+ *                   type: array
+ *                   items:
+ *                     type: object
+ *       '500':
+ *         description: Error listing models
+ *     security:
+ *       - session: []
+ *     tags: ["llm"]
+ */
+async function listModels(req: Request, res: Response) {
+    try {
+        const { baseUrl } = req.body;
+
+        // Use provided base URL or default from options
+        const ollamaBaseUrl = baseUrl || await options.getOption('ollamaBaseUrl') || 'http://localhost:11434';
+
+        // Call Ollama API to get models
+        const response = await axios.get(`${ollamaBaseUrl}/api/tags?format=json`, {
```

Comment:
I would propose to get rid of `axios` and just stick to the built-on `fetch` – from what I saw, we only have that depdency for the backend_script_api, where "axios" is marked as deprecated.

```

    /**
     * Axios library for HTTP requests. See {@link https://axios-http.com} for documentation
     * @deprecated use native (browser compatible) fetch() instead
     */
    axios: typeof axios;
```

Reply:
Moved what I found of `axios` to `fetch`...